gen_cdt_recipes.py

```diff
@@ -0,0 +1,189 @@
+import os
+import subprocess
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import tqdm
+import click
+from ruamel.yaml import YAML
+
+from cdt_config import (
+    LEGACY_CDT_PATH,
+    LEGACY_CUSTOM_CDT_PATH,
+    CDT_PATH,
+    CUSTOM_CDT_PATH,
+)
+
+yaml = YAML()
+
+
+def _make_cdt_recipes(*, extra, cdt_path, arch_dist_tuples, cdts, exec):
+    futures = {}
+    for arch, dist in arch_dist_tuples:
+        for cdt, cfg in cdts.items():
+            if cfg["custom"]:
+                continue
+
+            _extra = extra + " --recursive"
+
+            futures[exec.submit(
+                subprocess.run,
```

Comment:
Would it make sense to use the [`ProcessPoolExecutor`]( https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor )?

Reply:
It might or might not. My motivation to move to that is pretty limited. The `rpm.py` source is not really made to be used directly from python and this interface will work nicely with conda build at the command line if our changes ever get pushed upstream.

Reply:
issue here: https://github.com/conda-forge/cdt-builds/issues/3